test/TensorOneDTest.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "geopm/Exception.hpp"
+
+#include "TensorOneD.hpp"
+
+using geopm::TensorOneD;
+
+class TensorOneDTest : public ::testing::Test
+{
+    protected:
+        void SetUp();
+
+        TensorOneD one, two, three;
+};
+
+void TensorOneDTest::SetUp()
+{
+    one.set_dim(2);
+    two.set_dim(2);
+    three.set_dim(3);
+
+    one[0] = 1;
+    one[1] = 2;
+    two[0] = 3;
+    two[1] = 4;
+    three[0] = 0;
+    three[1] = 1;
+    three[2] = 1;
+}
+
+TEST_F(TensorOneDTest, test_sum)
+{
+    TensorOneD four = one + two;
+    EXPECT_EQ(4, four[0]);
+    EXPECT_EQ(6, four[1]);
+}
+
+TEST_F(TensorOneDTest, test_diff)
+{
+    TensorOneD four(one - two);
+    EXPECT_EQ(-2, four[0]);
+    EXPECT_EQ(-2, four[1]);
+}
+
+TEST_F(TensorOneDTest, test_dot)
+{
+    EXPECT_EQ(11, one * two);
+}
+
+TEST_F(TensorOneDTest, test_sigmoid)
+{
+    TensorOneD activations(5);
+
+    activations[0] = -log(1/0.1 - 1);
+    activations[1] = -log(1/0.25 - 1);
+    activations[2] = -log(1/0.5 - 1);
+    activations[3] = -log(1/0.75 - 1);
+    activations[4] = -log(1/0.9 - 1);
+
+    TensorOneD output = activations.sigmoid();
+
+    EXPECT_FLOAT_EQ(0.1, output[0]);
+    EXPECT_FLOAT_EQ(0.25, output[1]);
+    EXPECT_FLOAT_EQ(0.5, output[2]);
+    EXPECT_FLOAT_EQ(0.75, output[3]);
+    EXPECT_FLOAT_EQ(0.9, output[4]);
+}
+
+TEST_F(TensorOneDTest, test_copy)
+{
+    two = one;
+
+    // copy is successful
+    EXPECT_EQ(1, two[0]);
+    EXPECT_EQ(2, two[1]);
+
+    // copy is deep
+    two[0] = 9;
+    EXPECT_EQ(1, one[0]);
+    EXPECT_EQ(9, two[0]);
+}
+
+TEST_F(TensorOneDTest, test_input)
+{
+    TensorOneD x(3);
+    x.set_dim(4);
+    std::vector<float> vals = {8, 16};
+    x = TensorOneD(json11::Json(vals));
```

Comment:
A suggestion for an interface upgrade.
How would you like to add a constructor for `TensorOneD` which takes that vector in directly as a parameter?
```cpp
x = TensorOneD(vals);
```
We could have the json constructor call the vector constructor internally, since all the underlying logic is the same. What do you think?

Reply:
If possible, I think it would be a good idea to decouple these Tensor classes from json11, so that they're not dependent on a particular JSON library (since the concept they're encapsulating is independent of JSON parsing). I figure these classes take json11 objects as that's what might be convenient based on how the higher-level classes use them. I'm fine with keeping it as-is for now and revisiting this when reviewing the higher-level classes, to see if there's opportunity to isolate the JSON parsing to a single centralized place and decouple these classes from those details.